core/shared/src/main/scala-3/zio/prelude/Newtype.scala

```diff
@@ -72,98 +91,13 @@ abstract class Newtype[A] {
-            }
-
-          case _               =>
-            report.errorAndAbort(s"$refinementErrorHeader\nMust use literal value for macro.")
-        }
-
-      case None =>
-        assertionExpr.asTerm.underlying match {
-          // The user forgot to use the `inline` keyword
-          case Select(ident @ Ident(name), _) if ident.symbol.declarations.find(_.name == "assertion").exists { expr =>
-              expr.flags.is(Flags.Override)
-            } =>
-
-            val message = s"""$refinementErrorHeader
-
-We were unable to read your ${magenta(name)} assertion at compile-time.
-You must annotate ${yellow("def assertion")} with the ${yellow("inline")} keyword:
-
-    ${yellow("override ") + yellow(underlined("inline")) + yellow(" def assertion = ???")}.
-            """
-            report.errorAndAbort(message)
-
-          case Select(ident @ Ident(name), "assertion") if ident.tpe <:< TypeRepr.of[Newtype[_]]=> 
-            '{ $a.asInstanceOf[T] }
-
-          case other => 
-            val source = scala.util.Try(assertionExpr.asTerm.pos.sourceCode.get).getOrElse(assertionExpr.show)
-            val message = s"""$refinementErrorHeader
-
-We were unable to read your assertion at compile-time.
-
-You must either define your assertion directly or refer to other inline definitions:
-
-    ${yellow("override inline def assertion = greaterThan(10) && lessThan(100)")}.
-
-    or
-
-    ${yellow("inline def extracted = greaterThan(10) && lessThan(100)")}.
-    ${yellow("override inline def assertion = extracted")}.
-            """
-            report.errorAndAbort(message)
-        }
-    }
-  }
-
-  def makeMany_Impl[A: Type, T: Type](assertionExpr: Expr[Assertion[A]], a: Expr[A], as: Expr[Seq[A]])(using Quotes
-  ): Expr[NonEmptyChunk[T]] = {
-    import quotes.reflect.*
-
-    assertionExpr.value match {
-      case Some(assertion) =>
-        as match {
-          case Varargs(exprs) =>
-            val validated = exprs.map {
-              case LiteralUnlift(x) => assertion(x.asInstanceOf[A]).left.map(_.render(x.toString))
-              case _ => report.errorAndAbort(s"$refinementErrorHeader\nMust use literal value for macro.")
-            }
-
-            val (errors, _) = validated.partitionMap(identity)
-
-            // val rendered = ${dim("assertion =")} ${assertion}
-
-            if (errors.nonEmpty)
-               report.errorAndAbort(s"""$refinementErrorHeader
-${errors.mkString("\n")}
-""")
+abstract class SubtypeCustom[A] extends NewtypeCustom[A] {
+  type Base <: A
+}
 
-            '{ NonEmptyChunk($a, $as*).asInstanceOf[NonEmptyChunk[T]] }
+abstract class Newtype[A] extends NewtypeCustom[A] {
 
-          case _ =>
-              report.errorAndAbort(s"NO VARARGS!?")
-        }
-      
-      case None =>
-        report.errorAndAbort(s"NO REFINEMENT!?")
-    }
-  }
+  inline def validateInline(inline value: A): A =
```

Comment:
Btw, can this implementation be "upstreamed" from `Newtype` to `NewtypeCustom`? Are `inline` members overridable?

Reply:
No, sadly they are effectively final